rest/data/host.go

```diff
@@ -306,10 +306,11 @@ func fixProvisioningIntentHost(ctx context.Context, h *host.Host, instanceID str
 		// instance ID, there's nothing that can be done to fix it here.
 
 		// TODO (DEVPROD-6752): should log and return an error once all hosts
-		// roll over from the deploy. All intent hosts should be sending their
-		// EC2 instance ID. If they don't, it should fail provisioning and
-		// should not start the agent.
-		// msg := "intent host is running, but it did not provide an EC2 instance ID, which is required"
+		// roll over from the deploy and are providing their EC2 instance ID to
+		// this route. All intent hosts should be sending their EC2 instance ID.
+		// If they don't, it should fail provisioning and should not start the
+		// agent.
+		// msg := "intent host is up, but it did not provide an EC2 instance ID, which is required"
```

Comment:
Should we remove the above todo and uncomment the log here?

Reply:
We can log the warning but can't return the error yet because hosts started before this is deployed need to still provision without error (and they won't send their EC2 instance ID because `--provider` isn't set for pre-deploy hosts).

I uncommented the warning and adjusted the TODO.